### PR TITLE
add 2d rotation_angle

### PR DESCRIPTION
**Pull request by jw3126**

### CODE REVIEW

File: src/core_types.jl [hunk 1]
```diff
@@ -138,11 +138,18 @@ struct Angle2d{T} <: Rotation{2,T}
...
-Angle2d{T}(r::Angle2d) where {T} = Angle2d{T}(T(r.theta))
+Angle2d(r::Rotation) = Angle2d(rotation_angle(r))
+Angle2d{T}(r::Rotation) where {T} = Angle2d{T}(T(rotation_angle(r)))
 
 Base.one(::Type{A}) where {A<: Angle2d} = A(0)
 
+rotation_angle(rot::Angle2d) = rot.theta
+function rotation_angle(rot::RotMatrix{2})
+    c = rot[1,1]
+    s = rot[2,1]
```
Comment (tkoolen) [inline src/core_types.jl]: `@inbounds`?

Comment (jw3126) [reply src/core_types.jl]: ok why not :smile:

File: src/core_types.jl [hunk 2]
```diff
@@ -138,11 +138,18 @@ struct Angle2d{T} <: Rotation{2,T}
     theta::T
 end
 
-Angle2d(r::Angle2d) = r
-Angle2d{T}(r::Angle2d) where {T} = Angle2d{T}(T(r.theta))
+Angle2d(r::Rotation) = Angle2d(rotation_angle(r))
+Angle2d{T}(r::Rotation) where {T} = Angle2d{T}(T(rotation_angle(r)))
```
[the remark this answers is not in the document]
Comment (jw3126) [reply src/core_types.jl]: I think going `Rotation{3} -> Angle2d` is an exotic enough operation, that it should be spelling out explicitly as `Angle2d(rotation_angle(rot))`. 
As for the comment `# angle and axis introspection` I think it still applies, not sure what I should update here?

[the remark this answers is not in the document]
Comment (tkoolen) [reply src/core_types.jl]: > I think it still applies

Yeah, fair enough.

**Comment by tkoolen** [comment]
Great, thanks!

**Comment by tkoolen** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/21218

**Pull request merged**